### PR TITLE
Refactor and WebAssembly build

### PR DESCRIPTION
**Pull request by Shillaker**
Refactoring:

- Tidy up Python code
- Create Invoke tasks
- Single Dockerfile
- Simplify outputs to CSV file

Wasm experiment:

- WebAssembly build
- Faasm invocation tasks
- Data file upload tasks

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,122 @@
+build/
```
Comment (Shillaker) [inline .gitignore:1]: `.gitignore` was missing

File: tasks/run.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    """
+    Runs the native experiment
+    """
+    if not local:
+        print("Remote not yet implemented")
+        exit(1)
+
+    # Write header to results
+    with open(RESULTS_FILE, "w") as out_file:
+        out_file.write("Country,Threads,Run,Time(s)\n")
```
Comment (Shillaker) [inline tasks/run.py]: Writing results directly to a simple CSV makes them more readable and work very nicely with Python tools like Pandas

File: tasks/native.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        shell=True,
+        check=True,
+        cwd=BUILD_DIR,
+    )
+
+    run("cmake --build . --target all", shell=True, check=True, cwd=BUILD_DIR)
+
+
+@task
+def unzip(ctx):
```
Comment (Shillaker) [inline tasks/native.py:42]: This was done in the Dockerfile before so made it hard to do outside the container

File: tasks/run.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+FAASM_DATA_DIR = "faasm://covid"
+
+IMAGE_NAME = "experiment-covid"
+
+NATIVE_RESULTS_FILE = join(RESULTS_DIR, "covid_native.csv")
+WASM_RESULTS_FILE = join(RESULTS_DIR, "covid_wasm.csv")
+
+# Countries in order of size:
+# - Guam
+# - Malta
```
Comment (Shillaker) [inline tasks/run.py:32]: @csegarragonz can you remember what the list is? I can't work out what order they go in (smallest/ quickest to largest/ slowest)

**Review by csegarragonz — APPROVED** [review]

File: tasks/run.py (hunk 4 above)
Comment (csegarragonz) [reply tasks/run.py:32]: It was something like (increasing):
* Guam
* US Virgin Islands
* Iceland
* Malta
* Alaska (quite longer)

**Review by csegarragonz — APPROVED** [review]

**Pull request merged**